core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -966,6 +972,22 @@ private static SortOrder freshSortOrder(int orderId, Schema schema, SortOrder so
     return builder.build();
   }
 
+  private static Map<String, Long> indexAndValidateSnapshotTags(List<Snapshot> snapshots) {
+    Map<String, Long> snapshotTags = Maps.newHashMap();
+    for (Snapshot snap : snapshots) {
+      if (snap.tags() != null) {
+        for (String tag : snap.tags()) {
+          ValidationException.check(!snapshotTags.containsKey(tag),
+              "Tag is must be assigned to only 1 snapshot, but found tag %s in snapshots %s and %s",
```

Comment:
Nit: unnecessary "is"

Suggestion: "A tag can be assigned to only one snapshot, but ..."